rules/framework.bzl

```diff
@@ -159,22 +160,20 @@ def _get_virtual_framework_info(ctx, framework_files, compilation_context_fields
     # We need to map all the deps here - for both swift headers and others
     fw_dep_vfsoverlays = []
     for dep in transitive_deps + deps:
+        # Collect transitive headers. For now, this needs to include all of the
+        # transitive headers
+        if not CcInfo in dep:
```

Comment:
Could you explain why the order matters? Is it because an dep (except apple_framework_packaging targets) may return CcInfo, but won't return FrameworkInfo? Could we change it to something like
```
if CcInfo in dep:
  ...
if FrameworkInfo in dep:
 ...
```

Reply:
> Could you explain why the order matters? 
The code has a `continue` statement inside of the if block so it will skip ahead. 

> Could we change it to something like

I haven't tried that out yet but wouldn't be opposed to it. What are the benefits to changing the code to that?

Reply:
> The code has a continue statement inside of the if block so it will skip ahead.

I got that, but was  wondering why we need the headers of imported frameworks.

> What are the benefits to changing the code to that?

It might just be me, but the current logic is a little hard to understand. I had to know that if a dep doesn't return CcInfo, it won't return FrameworkInfo, either.

Reply:
Also we be better off just checking the providers like here https://github.com/bazel-ios/rules_ios/pull/333/files/5138482e5b1fcd470f76b793ef40e6ebc930cc73#r724580342 I don't believe there's exclusivity here but need to test it a bit more.